### PR TITLE
chore: upgrade to polkadot-v0.9.18

### PR DESCRIPTION
**Pull request by gregdhill**
Signed-off-by: Gregory Hill <gregorydhill@outlook.com>

[Polkadot Release Notes](https://github.com/paritytech/polkadot/releases/tag/v0.9.18)

## Notable Changes
- https://github.com/paritytech/substrate/pull/10825
- https://github.com/open-web3-stack/open-runtime-module-library/pull/714
- https://github.com/paritytech/cumulus/pull/1069

## Companion PRs

- [x] orml: https://github.com/open-web3-stack/open-runtime-module-library/pull/718
- [x] xcm-emulator: https://github.com/shaunxw/xcm-simulator/pull/26

## Dependencies

- [x] scale-info: 2.0.0
- [x] parity-scale-codec: 3.0.0

### CODE REVIEW

File: parachain/runtime/kintsugi/src/lib.rs [hunk 1]
```diff
@@ -904,6 +914,9 @@ impl orml_xtokens::Config for Runtime {
     type BaseXcmWeight = UnitWeightCost;
     type LocationInverter = <XcmConfig as Config>::LocationInverter;
     type MaxAssetsForTransfer = MaxAssetsForTransfer;
+    type MinXcmFee = ParachainMinFee;
+    type MultiLocationsFilter = Everything;
```
Comment (sander2) [inline parachain/runtime/kintsugi/src/lib.rs:918]: Let's change this to `[Kusama, Karura, Moonriver]`?

Comment (sander2) [reply parachain/runtime/kintsugi/src/lib.rs:918]: I also opened up an issue to additionally filter by asset: https://github.com/open-web3-stack/open-runtime-module-library/issues/724

File: parachain/runtime/kintsugi/src/lib.rs [hunk 2]
```diff
@@ -880,6 +881,15 @@ parameter_types! {
     pub const MaxAssetsForTransfer: usize = 2; // potentially useful to send both kint and kbtc at once
 }
 
+parameter_type_with_key! {
+    pub ParachainMinFee: |location: MultiLocation| -> u128 {
+        #[allow(clippy::match_ref_pats)] // false positive
+        match (location.parents, location.first_interior()) {
+            _ => u128::MAX,
+        }
+    };
```
Comment (sander2) [inline parachain/runtime/kintsugi/src/lib.rs:890]:
```suggestion
    // only used for transferring parachain tokens to other parachains using KSM as fee currency. Currently we do not support this, hence return MAX. See https://github.com/open-web3-stack/open-runtime-module-library/blob/cadcc9fb10b8212f92668138fc8f83dc0c53acf5/xtokens/README.md#transfer-multiple-currencies
    pub ParachainMinFee: |location: MultiLocation| -> u128 {
        u128::MAX
    };
```

File: parachain/runtime/kintsugi/src/lib.rs (hunk 1 above)
Comment (gregdhill) [reply parachain/runtime/kintsugi/src/lib.rs:918]: @sander2 are these locations already defined somewhere?

Comment (sander2) [reply parachain/runtime/kintsugi/src/lib.rs:920]: > @sander2 are these locations already defined somewhere?

No not yet, but should be something like this:
- Karura: `MultiLocation::new(1, X1(Parachain(2000))`
- Moonriver: `MultiLocation::new(1, X1(Parachain(2023))`
- Kusama: `MultiLocation::new(1, Here)`

Comment (gregdhill) [reply parachain/runtime/kintsugi/src/lib.rs]: @sander2 added please check, not sure if we should add an entry with `parents: 0`. See: https://github.com/open-web3-stack/open-runtime-module-library/blob/7e2f985d8ed49fe0d243e7ec8eb864cd9cc91d2a/xtokens/src/mock/para_relative_view.rs#L329

Comment (sander2) [reply parachain/runtime/kintsugi/src/lib.rs]: No that doesn't make sense, we should never use xtokens to send something to our own chain

File: parachain/runtime/kintsugi/src/lib.rs [hunk 3]
```diff
@@ -892,6 +905,17 @@ impl Convert<AccountId, MultiLocation> for AccountIdToMultiLocation {
...
 }
 
+match_type! {
+    pub type ParentOrParachains: impl Contains<MultiLocation> = {
+        // Kusama
+        MultiLocation { parents: 1, interior: X1(Junction::AccountId32 { .. }) } |
+        // Karura
+        MultiLocation { parents: 1, interior: X2(Parachain(2000), Junction::AccountId32 { .. }) } |
+        // Moonriver
+        MultiLocation { parents: 1, interior: X2(Parachain(2023), Junction::AccountId32 { .. }) }
```
Comment (sander2) [inline parachain/runtime/kintsugi/src/lib.rs:915]: Moonriver uses `AccountKey20`

File: parachain/runtime/runtime-tests/src/relaychain/kusama_test_net.rs [hunk 4]
```diff
@@ -62,7 +62,7 @@ fn default_parachains_host_configuration() -> HostConfiguration<BlockNumber> {
         max_upward_queue_size: 1024 * 1024,
         max_downward_message_size: 1024,
         ump_service_total_weight: 4 * 1_000_000_000,
-        max_upward_message_size: 1024 * 1024,
+        max_upward_message_size: 5 * 1024,
```
Comment (sander2) [inline parachain/runtime/runtime-tests/src/relaychain/kusama_test_net.rs:65]: Let's set to 50 * 1024? That is what kusama currently has it set to

File: parachain/runtime/testnet/src/lib.rs [hunk 5]
```diff
@@ -909,6 +910,15 @@ parameter_types! {
     pub const MaxAssetsForTransfer: usize = 2; // potentially useful to send both kint and kbtc at once
 }
 
+parameter_type_with_key! {
+    pub ParachainMinFee: |location: MultiLocation| -> u128 {
```
Comment (sander2) [inline parachain/runtime/testnet/src/lib.rs:914]: copy the one from kintsugi

File: parachain/runtime/interlay/src/lib.rs [hunk 6]
```diff
@@ -843,6 +844,15 @@ parameter_types! {
     pub const MaxAssetsForTransfer: usize = 2;
 }
 
+parameter_type_with_key! {
+    pub ParachainMinFee: |location: MultiLocation| -> u128 {
```
Comment (sander2) [inline parachain/runtime/interlay/src/lib.rs:848]: copy the one from kintsugi for consistency

**Review by sander2 — APPROVED** [review]

**Pull request merged**